src/main/scala/is/hail/expr/FunctionRegistry.scala

```diff
@@ -237,10 +226,113 @@ object FunctionRegistry {
           v = args(2).run(ec)();
 
           _ = (if (v == null)
-            fatal(s"Argument evaluated to missing in call to aggregator $name"));
+            fatal(s"Argument evaluated to missing in call to aggregator $name"))
+
+        ) yield aggregator.ctor(g, v.asInstanceOf[v])
+
+      case aggregator: GroupByAggregatorFun[t] =>
+        // val downstreamBodyFn = CM.ret(Code._null[AnyRef]).runWithDelayedValues(Seq(), EvalContext());
+
+        val Lambda(_, keyParam, keyBody) = args(1)
+        val TFunction(Seq(keyParamType), _) = argTypes(1)
+
+        val Lambda(_, downstreamParam, downstreamBody) = args(2)
+        val TFunction(Seq(downstreamParamType), _) = argTypes(2)
+
+        val kbc = keyBody.compile()
 
-          aggregationResultThunk <- addAggregation(args(0), aggregator.ctor(g, v.asInstanceOf[v]));
+        val a0 = args(0)
+
+        val a0type = a0.`type`
+
+        for (
+          (downstreamBodyCompiled, downstreamAggregator) <- downstreamBody.downstreamAggregator();
+
+          ec <- ec();
+          (idx, localA) <- ecNewPosition();
+
+          bodyST = a0type match {
+            case tagg: TAggregable => tagg.symTab
+            case _ => ec.st
```

Comment:
Dunno. @cseed? I copied this from other aggregators with lambdas. I think this case should be an error. You shouldn't call this method on anything that isn't agreeable.